model/implementation/DeliveryExecutionStateService.php

```diff
@@ -266,6 +267,8 @@ public function finishExecution(DeliveryExecution $deliveryExecution, $reason =
      */
     public function finish(DeliveryExecution $deliveryExecution, $reason = null)
     {
+        $eventManager = $this->getServiceManager()->get(EventManager::SERVICE_ID);
+        $eventManager->trigger(new DeliveryExecutionFinished($deliveryExecution));
```

Comment:
Maybe you can consider event triggering after state has been actually successfully changed?